src/oph/ehoks/palaute/tyoelama.clj

```diff
@@ -190,83 +162,93 @@
         (dissoc :internal-kyselytyyppi :jakson-yksiloiva-tunniste)
         (utils/remove-nils)
         (ddb/sync-jakso-herate!))))
 
 (defn create-and-save-arvo-vastaajatunnus!
   [tep-palaute]
-  (let [opiskeluoikeus (koski/get-opiskeluoikeus!
-                         (:opiskeluoikeus-oid tep-palaute))]
-    (if-not opiskeluoikeus
-      (log/warnf
-        "Opiskeluoikeus not found for palaute %d, skipping processing"
-        (:id tep-palaute))  ; FIXME: create tapahtuma and update state
-      (let [koulutustoimija (palaute/koulutustoimija-oid! opiskeluoikeus)
-            alkupvm         (next-niputus-date (:loppupvm tep-palaute))
-            request-id      (str (UUID/randomUUID))
-            suoritus        (find-first suoritus/ammatillinen?
-                                        (:suoritukset opiskeluoikeus))
-            arvo-request    (arvo/build-jaksotunnus-request-body
-                              tep-palaute
-                              (normalize-string
-                                (:tyopaikan-nimi tep-palaute))
-                              opiskeluoikeus
-                              request-id
-                              koulutustoimija
-                              (palaute/toimipiste-oid! suoritus)
-                              suoritus
-                              (str alkupvm))
-            arvo-response   (arvo/create-jaksotunnus arvo-request)
-            tunnus          (:tunnus arvo-response)]
-        (jdbc/with-db-transaction
-          [tx db/spec]
-          (try
-            (palaute/save-arvo-tunniste!
-              tx tep-palaute arvo-response {:arvo-response arvo-response})
-            (sync-jakso-to-heratepalvelu!
-              tx tep-palaute opiskeluoikeus request-id tunnus)
+  {:pre [(:vastuullinen-tyopaikka-ohjaaja-nimi tep-palaute)
+         (:tyopaikan-nimi tep-palaute) (:tyopaikan-y-tunnus tep-palaute)]}
+  (if-let [opiskeluoikeus (koski/get-opiskeluoikeus!
+                            (:opiskeluoikeus-oid tep-palaute))]
+    (let [suoritus (find-first suoritus/ammatillinen?
+                               (:suoritukset opiskeluoikeus))
+          ctx {:opiskeluoikeus  opiskeluoikeus
+               :suoritus        suoritus
+               :koulutustoimija (palaute/koulutustoimija-oid! opiskeluoikeus)
+               :toimipiste      (palaute/toimipiste-oid! suoritus)
+               :niputuspvm      (next-niputus-date (date/now))}]
+      ; The following attributes are required for TPO-nippu.
+      (assert (and (:koulutustoimija ctx) (:niputuspvm ctx) (:suoritus ctx)))
+      (jdbc/with-db-transaction
+        [tx db/spec]
+        (let [keskeytymisajanjaksot
+              (oht/get-keskeytymisajanjaksot!
+                tx {:oht-id (:hankkimistapa-id tep-palaute)})
+              nippu (nippu/build-tpo-nippu-for-heratepalvelu
+                      ctx tep-palaute keskeytymisajanjaksot)]
+          (if (= (:kasittelytila nippu) "ei_niputettu")
```

Comment:
Myös tässä on tosi pitkä `(if)`, jossa raskas ensimmäinen lause, pitäisi kääntää toisin päin.  vakavuus: pyyntö